File: data/gpt-4o/readable_format_HPEC/cpp/file235.cpp
#include <stdio.h>
#define DATA_TYPE double

DATA_TYPE c[100][100];
DATA_TYPE w[100][100];
DATA_TYPE sumC[100][100][100];

void kernel_dynprog(int tsteps, int length, DATA_TYPE output)
{
    int i, j, iter, k;

    output = 0;

    for (iter = 0; iter < tsteps; iter++) {
        for (i = 0; i < length; i++) {
            for (j = 0; j < length; j++) {
                c[j][i] = 0;
            }
        }

        for (i = 0; i < length-1; i++) {
            for (j = i+1; j < length; j++) {
                sumC[i][j][i] = 0;
                for (k = i+1; k < j-1; k++) {
                    sumC[k][j][i] = sumC[k-1][j][i] + c[k][i] + c[j][k];
                }
                c[j][i] = sumC[j-1][j][i] + w[j][i];
            }
        }
        output = output + c[length-1][0];
    }
}
int main() {
    int tsteps = 10;
    int length = 100;
    DATA_TYPE output = 0.0;
    kernel_dynprog(tsteps, length, output);
    return 0;
}

File: data/deepseek_coder_base_6.7B/readable_format_HPEC/fortran/file235.f90
subroutine kernel_dynprog(tsteps , length, c, w, sumC, output)
implicit none

DATA_TYPE, dimension(length, length) :: w, c
DATA_TYPE, dimension(length, length, length) :: sumC
integer :: i, j, iter, k
integer :: length, tsteps
DATA_TYPE :: output

!$pragma scop
output = 0

do iter = 1, _PB_TSTEPS
do i = 1, _PB_LENGTH
do j = 1, _PB_LENGTH
c(j, i) = 0
end do
end do

do i = 1, _PB_LENGTH - 1
do j = i + 1, _PB_LENGTH
sumC(i, j, i) = 0
do k = i + 1, j - 1
sumC(k, j, i) = sumC(k - 1, j, i) + c(k, i) + c(j, k)
end do
c(j, i) = sumC(j - 1, j, i) + w(j, i)
end do
end do
output = output + c(_PB_LENGTH, 1)
end do
!$pragma endscop
end subroutine
